leader/undirected_graph/sync_casteigts_metivier_robson_zemmari/sync_casteigts_metivier_robson_zemmari.go

```diff
@@ -0,0 +1,551 @@
+package sync_casteigts_metivier_robson_zemmari
+
+import (
+	"encoding/json"
+	"fmt"
+	"log"
+	"math/bits"
+	"math/rand"
+	"time"
+
+	"github.com/krzysztof-turowski/distributed-framework/lib"
+)
+
+// Run the Casteigts-Métivier-Robson-Zemmari deterministic leader election algorithm on a random network
+func Run(n int, p float64) (int, int) {
+	nodes, synchronizer := buildSynchronizedRandomConnectedGraph(n, p)
+	rng := rand.New(rand.NewSource(time.Now().UnixNano()))
+	identifiers := rng.Perm(n)
+	for i, node := range nodes {
+		log.Println("Node", node.GetIndex(), "about to run")
+		go run(node, 2*(identifiers[i]+1)) // Every node is assigned a unique identifier of O(log(n)) bits
+	}
+	synchronizer.Synchronize(0)
+	check(nodes)
+	return synchronizer.GetStats()
+}
+
+func buildSynchronizedRandomConnectedGraph(n int, p float64) ([]lib.Node, lib.Synchronizer) {
+	possibleEdges := (n * (n - 1)) / 2
+	m := int(float64(possibleEdges) * p)
+	m = max(m, n-1)
+	m = min(m, possibleEdges)
+	weightedNodes, synchronizer := lib.BuildSynchronizedRandomConnectedWeightedGraph(n, m, 1, lib.GetGenerator())
+	nodes := make([]lib.Node, len(weightedNodes))
+	for i, weightedNode := range weightedNodes {
+		nodes[i] = weightedNode
+	}
+	return nodes, synchronizer
+}
```

Comment:
Why don't you use functions from lib_graph.go? Please use them, and/or add their extensions there.